### ISSUE TITLE
Don’t warn that function references arguments which are not passed if set is used

### ISSUE THREAD

**Issue by mina86** [koalaman/shellcheck#690]
At times I use function’s positional arguments to store local variables. This is useful since POSIX shells do not have `local` directive. An example code might look something like:

```shell
foo() {
    set -- 'foo * bar'
    printf '%s\n' "$1"
}
foo
```

Sadly, this leads to noisy shellcheck output:

```
In a.sh line 1:
foo() {
^-- SC2120: foo references arguments, but none are ever passed.


In a.sh line 5:
foo
^-- SC2119: Use foo "$@" if function's $1 should mean script's $1.
```

It would be fantastic, if shellcheck detected that set builtin is used inside of the function and in those cases not issued the warnings.

**Comment by bittorf** [comment]
this seems fixed - so please close
